go/ql/lib/ext/github.com.go-jose.go-jose.model.yml

```diff
@@ -0,0 +1,14 @@
+extensions:
+  - addsTo:
+      pack: codeql/go-all
+      extensible: packageGrouping
+    data:
+      - ["go-jose", "github.com/go-jose/go-jose"]
+      - ["go-jose", "gopkg.in/square/go-jose"]
+      - ["go-jose", "github.com/square/go-jose"]
```

Comment:
Does this also handle `"gopkg.in/square/go-jose.v2"`? Just double checking that the `.v2` is ok.

Reply:
Yes, [this regex](https://github.com/github/codeql/blob/58689c90fbed2abc5003fa82206f7e079f932b09/go/ql/lib/semmle/go/Packages.qll#L36-L42) matches `.v2` or `/v2`, so then [this predicate](https://github.com/github/codeql/blob/58689c90fbed2abc5003fa82206f7e079f932b09/go/ql/lib/semmle/go/Packages.qll#L21-L27) removes either, and that is used [here](https://github.com/github/codeql/blob/58689c90fbed2abc5003fa82206f7e079f932b09/go/ql/lib/semmle/go/dataflow/ExternalFlow.qll#L438) on the imported package name to match it against the string given in the MaD yml file.

[This test](https://github.com/github/codeql/blob/58689c90fbed2abc5003fa82206f7e079f932b09/go/ql/test/query-tests/Security/CWE-798/jwt.go#L29) imports it in that way and it still passes on this PR.

Reply:
Fun fact: I found out recently that the `.v2` format is only for `gopkg.in`. I guess that site used that syntax before go modules were invented and it was easier to accept it than to make everyone change their imports.